frontend/packages/dev-console/src/components/pipelines/modals/common/PipelineSecretSection.tsx

```diff
@@ -1,66 +1,57 @@
+import { SecretKind } from '@console/internal/module/k8s';
+import { getSecretAnnotations } from '../../../../utils/pipeline-utils';
 import SecretForm from './SecretForm';
 import SecretsList from './SecretsList';
+import { secretFormDefaultValues, isDuplicate } from './utils';
+import { useFetchSecrets } from './hooks';
 
 import './PipelineSecretSection.scss';
 
-const initialValues = {
-  secretName: '',
-  annotations: { key: SecretAnnotationId.Image, value: '' },
-  type: SecretType.dockerconfigjson,
-  formData: {},
-};
-
 type PipelineSecretSectionProps = {
   namespace: string;
 };
 
 const PipelineSecretSection: React.FC<PipelineSecretSectionProps> = ({ namespace }) => {
-  const [secretOpenField] = useField<boolean>('secretOpen');
-  const { setFieldValue } = useFormikContext<FormikValues>();
+  const {
+    setFieldValue,
+    values: { newSecrets, secretOpen, secretForm },
+    setFieldTouched,
+    setFieldError,
+  } = useFormikContext<FormikValues>();
+  const existingSecrets: SecretKind[] = useFetchSecrets(namespace);
 
-  const handleSubmit = (values, actions) => {
-    actions.setSubmitting(true);
-    const newSecret = {
-      apiVersion: SecretModel.apiVersion,
-      kind: SecretModel.kind,
-      metadata: {
-        name: values.secretName,
-        namespace,
-        annotations: getSecretAnnotations(values.annotations),
-      },
-      type: values.type,
-      stringData: values.formData,
-    };
-    k8sCreate(SecretModel, newSecret)
-      .then((resp) => {
-        actions.setSubmitting(false);
-        setFieldValue(secretOpenField.name, false);
-        associateServiceAccountToSecret(resp, namespace);
-      })
-      .catch((err) => {
-        actions.setSubmitting(false);
-        setFieldValue(secretOpenField.name, false);
-        actions.setStatus({ submitError: err.message });
-      });
+  const resetSecretForm = () => {
+    setFieldTouched(`secretForm`, false);
+    setFieldValue(`secretForm`, secretFormDefaultValues);
+    setFieldValue(`secretOpen`, false);
   };
 
-  const handleReset = (values, actions) => {
-    actions.resetForm({ values: initialValues, status: {} });
-    setFieldValue(secretOpenField.name, false);
+  const handleSubmit = () => {
+    if (isDuplicate(existingSecrets, secretForm.secretName)) {
+      setFieldError(`secretForm.secretName`, 'Secret name already exists.');
+    } else if (isDuplicate(newSecrets, secretForm.secretName)) {
+      setFieldError(`secretForm.secretName`, 'Secret name already added.');
+    } else {
+      const newSecret: SecretKind = {
+        apiVersion: SecretModel.apiVersion,
+        kind: SecretModel.kind,
+        metadata: {
+          name: secretForm.secretName,
+          namespace,
+          annotations: getSecretAnnotations(secretForm.annotations),
+        },
+        type: secretForm.type,
+        stringData: secretForm.formData,
+      };
+      setFieldValue(`newSecrets`, [...newSecrets, newSecret]);
+      resetSecretForm();
```

Comment:
```suggestion
      k8sCreate(SecretModel, newSecret, { queryParams: { dryRun: 'All' } })
        .then(() => {
          setFieldValue(`newSecrets`, [...newSecrets, newSecret]);
          resetSecretForm();
          setStatus({ submitError: null });
        })
        .catch((err) => {
          setStatus({ submitError: err.message });
        });
```
It's not perfect, the backend seems not to have the same restrictions as the frontend shows. But it's something for the time being.

Reply:
As discussed, save the error locally using `useState` (instead of `setStatus`) and render a pf inline Alert component just above [the buttons in the SecretForm](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/pipelines/modals/common/SecretForm.tsx#L106)

Reply:
Should we disable the Check button if there's an error in the secret form ?